README.md

```diff
@@ -0,0 +1,523 @@
+
+```coffeescript
+  # Yes
+  if true
+    ...
+  else
+    ...
+
+  # No
+  if true then ...
+  else ...
+```
+
+<a name="looping_and_comprehensions"></a>
+## Looping and Comprehensions
+
+For arrays, try to use native Array.prototype methods
+
+```coffeescript
+  # No
+  result = (item.name for item in array)
+
+  # No
+  results = []
+  for item in array
+    results.push item.name
+
+  # Yes
+  result = array.map((item) -> item.name)
+```
+
+To filter:
+
+```coffeescript
+result = items.filter((item) -> item.name is 'test')
+```
+
+To iterate over the keys and values of objects:
+
+```coffeescript
+object =
+  one: 1
+  two: 2
+alert("#{key} = #{value}") for key, value of object
+```
+
+<a name="extending_native_objects"></a>
+## Extending Native Objects
+
+Do not modify native objects.
+
+For example, do not modify `Array.prototype` to introduce `Array#forEach`.
+
+<a name="exceptions"></a>
+## Exceptions
+
+Do not suppress exceptions.
+
+<a name="annotations"></a>
+## Annotations
+
+Use annotations when necessary to describe a specific action that must be taken against the indicated block of code.
+
+Write the annotation on the line immediately above the code that the annotation is describing.
+
+The annotation keyword should be followed by a colon and a space, and a descriptive note.
+
+```coffeescript
+  # FIXME: The client's current state should *not* affect payload processing.
+  resetClientState()
+  processPayload()
+```
+
+If multiple lines are required by the description, indent subsequent lines with two spaces:
+
+```coffeescript
+  # TODO: Ensure that the value returned by this call falls within a certain
+  #   range, or throw an exception.
+  analyze()
+```
+
+Annotation types:
+
+- `TODO`: describe missing functionality that should be added at a later date
+- `FIXME`: describe broken code that must be fixed
+- `OPTIMIZE`: describe code that is inefficient and may become a bottleneck
+- `HACK`: describe the use of a questionable (or ingenious) coding practice
+- `REVIEW`: describe code that should be reviewed to confirm implementation
+
+If a custom annotation is required, the annotation should be documented in the project's README.
+
+<a name="miscellaneous"></a>
+## Miscellaneous
+
+`and` is preferred over `&&`.
+
+`or` is preferred over `||`.
+
+`is` is preferred over `==`.
+
+`isnt` is preferred over `!=`.
+
+`not` is preferred over `!`.
+
+`or=` should be used when possible:
+
+```coffeescript
+temp or= {} # Yes
+temp = temp || {} # No
+```
+
+Prefer shorthand notation (`::`) for accessing an object's prototype:
+
+```coffeescript
+Array::slice # Yes
+Array.prototype.slice # No
+```
+
+Prefer `@property` over `this.property`.
+
+```coffeescript
+return @property # Yes
+return this.property # No
+```
+
+However, avoid the use of **standalone** `@`:
+
+```coffeescript
+return this # Yes
+return @ # No
+```
+
+Avoid `return` where not required, unless the explicit return increases clarity.
```

Comment:
We might consider also adding something here about explicitly returning long or multi-line chunks of code to improve clarity